### PR TITLE
[FLINK-16589][table-planner-blink] Split code for AggsHandlerCodeGene…

### PR DESCRIPTION
**Pull request by libenchao**
…rator

## What is the purpose of the change

Split code for AggsHandlerCodeGenerator

## Brief change log

Split code for AggsHandlerCodeGenerator

## Verifying this change

This change added tests and can be verified as follows:

- AggrateITCase.testSplitAggsHandler

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know) 
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/GenerateUtils.scala [hunk 1]
```diff
@@ -204,39 +204,47 @@ object GenerateUtils {
...
         val binaryRowWriter = className[BinaryRowWriter]
         val typeTerm = clazz.getCanonicalName
+        ctx.addReusableMember(s"final $typeTerm $recordTerm = new $typeTerm(${rt.getFieldCount});")
+        ctx.addReusableMember(
+          s"final $binaryRowWriter $writerTerm = new $binaryRowWriter($recordTerm);")
         s"""
-           |final $typeTerm $recordTerm = new $typeTerm(${rt.getFieldCount});
-           |final $binaryRowWriter $writerTerm = new $binaryRowWriter($recordTerm);
+           |$recordTerm = new $typeTerm(${rt.getFieldCount});
+           |$writerTerm = new $binaryRowWriter($recordTerm);
```
Comment (wuchong) [inline flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/GenerateUtils.scala:234]: If we always put it in member variables, we don't need the initialization statements anymore?

Comment (libenchao) [reply flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/GenerateUtils.scala:237]: the record maybe not be reused, for example, in `ExprCodeGenerator.generateResultExpression` and in `ProjectionCodeGenerator.generateProjectionExpression`

**Comment by libenchao** [comment]
@wuchong I've rebased to master and resolved conflicts.

File: flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/GenerateUtils.scala (hunk 1 above)
Comment (wuchong) [reply flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/GenerateUtils.scala:237]: Thanks for the explanation.

File: flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/CodeGeneratorContext.scala [hunk 2]
```diff
@@ -210,10 +211,14 @@ class CodeGeneratorContext(val tableConfig: TableConfig) {
     */
   def reuseMemberCode(): String = {
     val result = reusableMemberStatements.mkString("\n")
-    if (isCodeSplit) {
+    if (isCodeSplitMap.values.exists(identity)) {
```
Comment (wuchong) [inline flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/CodeGeneratorContext.scala:214]: Could we use `isCodeSplitMap.nonEmpty`?

**Comment by libenchao** [comment]
@wuchong Thanks for the review, I've rebased to the master and addressed your comments.

**Comment by wuchong** [comment]
Could you open another pull request againt the release-1.11 branch ? @libenchao

**Comment by libenchao** [comment]
> Could you open another pull request againt the release-1.11 branch ? 

Yes, of course. 
I have one small question, as our 1.11 release manager said in ML[1], can we still merge pr into release-1.11 branch now? 

[1] http://apache-flink-mailing-list-archive.1008284.n3.nabble.com/ANNOUNCE-Apache-Flink-1-11-0-release-candidate-2-tc42620.html

**Comment by wuchong** [comment]
@libenchao , no , we will merge into release-1.11 when RC2 is out. That means this fix maybe released in 1.11.1, but if RC2 is cancelled, this fix can still go into 1.11.0.

File: flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/runtime/stream/sql/AggregateITCase.scala [hunk 3]
```diff
@@ -1334,4 +1334,32 @@ class AggregateITCase(
...
+  @Test
+  def testSplitAggsHandler(): Unit = {
+
+    val t = env.fromCollection(TestData.smallTupleData3)
+      .toTable(tEnv, 'a, 'b, 'c)
+    tEnv.createTemporaryView("MyTable", t)
+
+    tEnv.getConfig.setMaxGeneratedCodeLength(1)
+
+    val columnNumber = 100
```
Comment (wuchong) [inline flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/runtime/stream/sql/AggregateITCase.scala:1347]: I would like to update this a bit. I tried to revert the changes and this test is still passed. That means this test can't reproduce the 64K problem. I changed to number up to `500` and reproduced this problem. Besides, I would like to remove `tEnv.getConfig.setMaxGeneratedCodeLength(1)` to make sure it works well with out-of-box configuration.

File: flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/runtime/stream/sql/AggregateITCase.scala [hunk 4]
```diff
@@ -1334,4 +1334,32 @@ class AggregateITCase(
     val expected = Seq("3,29.39,tom123@gmail.com")
     assertEquals(expected.sorted, sink.getRetractResults.sorted)
   }
+
+  @Test
+  def testSplitAggsHandler(): Unit = {
```
Comment (wuchong) [inline flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/runtime/stream/sql/AggregateITCase.scala:1339]: Could you rename to `testAggregationCodeSplit` ?  We have a `SplitAggregateITCase` test which may be confuse users the naming.

**Comment by libenchao** [comment]
@wuchong got it. will prepare another pr against release-1.11 branch.

**Comment by libenchao** [comment]
@flinkbot run azure

File: flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/runtime/stream/sql/AggregateITCase.scala (hunk 3 above)
Comment (libenchao) [reply flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/runtime/stream/sql/AggregateITCase.scala:1347]: OK, will change it.

**Comment by libenchao** [comment]
@wuchong Thanks for the review, will merge once CI gives green result.

**Comment by libenchao** [comment]
tests passed in my CI: https://dev.azure.com/libenchao/flink.benchao/_build/results?buildId=19&view=results
merging
